LanguageFeatures/Patterns/variable_A02_t02.dart

```diff
@@ -73,7 +111,28 @@ String testMap(Map m) {
   }
 }
 
-String testObject(Shape shape) {
+String testMap2(Map m) {
+  return switch (m) {
+    case {1: a} => "{1: $a}";
+    case {2: b} => "{2: $b}";
+    default => "default";
+  };
+}
+
+String testMap3(Map m) {
+  if (m case {1: a}) {
+    a.expectStaticType<Exactly<int>>();
```

Comment:
OK, please search for `expectStaticType`. ;-)

I think it's a good idea to test for the static type. In this case it's clearly `dynamic` (because `Map` means `Map<dynamic, dynamic>`, by instantiation-to-bound). But I don't know whether it's possible to use `expectStaticType` to detect `dynamic`.

It's testing for subtype relationships in a function type in order to get both "less equal" and "greater equal", but that won't see the difference between `dynamic` and any other top type. The real test for `dynamic` would be that `if (b) myTarget.whatever();` is not a compile-time error, where `b` is a top level variable initialized to `false` (such that we don't get in trouble with `invariant_boolean`, and the compiler doesn't delete the if statement entirely).